src/flow/assembly_lmh_old.hh

```diff
@@ -28,598 +47,688 @@
-     * and reconstructs the full solution on the element.
-     * Currently used only for seepage BC.
-     */
-    void load_local_system(const DHCellAccessor& dh_cell)
-    {
-        // do this only once per element
-        if(bc_fluxes_reconstruted)
-            return;
+    /// Destructor.
+    virtual ~MHMatrixAssemblyLMH() {}
 
-        // possibly find the corresponding local system
-        auto ls = ad_->seepage_bc_systems.find(dh_cell.elm_idx());
-        if (ls != ad_->seepage_bc_systems.end())
-        {
-            arma::vec schur_solution = ad_->p_edge_solution.get_subvec(loc_schur_.row_dofs);            
-            // reconstruct the velocity and pressure
-            ls->second.reconstruct_solution_schur(schur_offset_, schur_solution, reconstructed_solution_);
+    /// Initialize auxiliary vectors and other data members
+    void initialize(ElementCacheMap *element_cache_map) {
+        //this->balance_ = eq_data_->balance_;
+        this->element_cache_map_ = element_cache_map;
 
-            postprocess_velocity(dh_cell, reconstructed_solution_);
+        fe_ = std::make_shared< FE_P_disc<dim> >(0);
+        fe_values_side_.initialize(*this->quad_low_, *fe_, update_normal_vectors);
 
-            bc_fluxes_reconstruted = true;
-        }
+        fe_values_.initialize(quad_rt_, fe_rt_, update_values | update_JxW_values | update_quadrature_points);
+
+        ndofs_ = fe_values_.n_dofs();
+        qsize_ = fe_values_.n_points();
+        this->use_dirichlet_switch_ = true;
+
+        // local numbering of dofs for MH system
+        // note: this shortcut supposes that the fe_system is the same on all elements
+        // TODO the function should be DiscreteSpace.fe(ElementAccessor)
+        // and correct form fe(ad_->dh_->own_range().begin()->elm()) (see documentation of DiscreteSpace::fe)
+        auto fe = eq_data_->dh_->ds()->fe()[Dim<dim>{}];
+        FESystem<dim>* fe_system = dynamic_cast<FESystem<dim>*>(fe.get());
+        eq_data_->loc_side_dofs[dim-1] = fe_system->fe_dofs(0);
+        eq_data_->loc_ele_dof[dim-1] = fe_system->fe_dofs(1)[0];
+        eq_data_->loc_edge_dofs[dim-1] = fe_system->fe_dofs(2);
+
+        // reconstructed vector for the velocity and pressure
+        // length = local Schur complement offset in LocalSystem
+        eq_data_->schur_offset_[dim-1] = eq_data_->loc_edge_dofs[dim-1][0];
+        reconstructed_solution_.zeros(eq_data_->schur_offset_[dim-1]);
     }
 
 
-    /// Saves the local system to a map: element index -> LocalSystem.
-    /// Currently used only for seepage BC.
-    void save_local_system(const DHCellAccessor& dh_cell)
-    {
-        // for seepage BC, save local system
-        if(save_local_system_)
-            ad_->seepage_bc_systems[dh_cell.elm_idx()] = loc_system_;
+    /// Setter of use_dirichlet_switch data member
+    inline void set_dirichlet_switch(bool ds) {
+        this->use_dirichlet_switch_ = ds;
```

Comment:
Dirichlet switch was a hack to turn off a part of assembly, we should rather separate that part into separate methods, have separate assembly class for reconstruct assembly, but derived from the full assembly and reusing all methods, but avoid that particular method.